README.tpl.md

```diff
@@ -0,0 +1,133 @@
+
+For more information about Scalastyle and Scapegoat rules, please consult the upstream documentation:
+- Scalastyle - http://www.scalastyle.org/rules-1.0.0.html
+- Scapegoat - https://github.com/sksamuel/scapegoat/tree/v1.3.8
+
+
+# Set-up
+Download the latest [release](https://github.com/mwz/sonar-scala/releases) jar into your SonarQube plugins folder `/opt/sonarqube/extensions/plugins` and restart SonarQube either manually or using the update center.
+
+For an out-of-the-box setup, you can use my docker-compose recipe or a docker image with SonarQube which contains bundled sonar-scala plugin. Please see [mwz/sonar-scala-docker](https://github.com/mwz/sonar-scala-docker) for more details.
+
+For automating the analysis of your Scala projects, check out my sbt plugin [mwz/sbt-sonar](https://github.com/mwz/sbt-sonar).
+
+Also, see the [examples](https://github.com/mwz/sonar-scala/tree/master/examples) directory, which includes sample projects for sbt, Gradle and Maven along with basic instructions on how to execute SonarQube analysis for each of those projects.
+
+
+# Sonar-scanner properties
+The plugin exposes the following properties which can be passed to sonar-scanner when running an analysis:
+- **sonar.sources** - Scala source directory relative to the root of your project (usually `src/main/scala`)
+- **sonar.scala.version** (optional) - defines the version of Scala used in your project (requires the `{major}.{minor}` versions and the patch version is ignored, defaults to `2.12`)
+- **sonar.scala.scoverage.reportPath** (optional) - relative path to the scoverage report (defaults to `target/scala-${sonar.scala.version}/scoverage-report/scoverage.xml`)
+- **sonar.scala.scalastyle.disable** (optional) - disables the Scalastyle sensor from being executed on your sources (defaults to `false`)
+- **sonar.scala.scapegoat.reportPath** (optional) - relative path to the scapegoat report (defaults to `target/scala-${sonar.scala.version}/scapegoat-report/scapegoat.xml`)
+- **sonar.scala.scapegoat.disable** (optional) - disables the Scapegoat sensor from being executed on your sources (defaults to `false`)
+
+See an example usage:
+```bash
+sonar-scanner -Dsonar.projectName=test \
+              -Dsonar.projectKey=test \
+              -Dsonar.sources=src/main/scala \
+              -Dsonar.sourceEncoding=UTF-8 \
+              -Dsonar.scala.version=2.12 \
+              -Dsonar.scoverage.reportPath=target/scala-2.12/scoverage-report/scoverage.xml \
+              -Dsonar.scapegoat.reportPath=target/scala-2.12/scapegoat-report/scapegoat.xml
+```
+
+
+# Compatibility with SonarQube
+SonarQube | sonar-scala
+----------|----------
+7.4       | [7.1](https://github.com/mwz/sonar-scala/releases/tag/v7.1.0), [7.2](https://github.com/mwz/sonar-scala/releases/tag/v7.2.0)
```

Comment:
Maybe here we should make something like **7.x**_(latest 7.2)_ and use the `version` variable.
However, since **7.0 ** it maye cause confusion but IMHO the table is clear enough.

Reply:
I also thought about it and like you said it could be confusing, so I decided to list here explicitly all the versions.